docs/modules/ROOT/pages/documentation/extensions/grpc/README.adoc

```diff
@@ -0,0 +1,415 @@
+                    </goals>
+                </execution>
+            </executions>
+        </plugin>
+    </plugins>
+</build>
+----
+
+* The `os-maven-plugin` extension is used to identify OS properties used during generation of stubs classes.
+* The `protobuf-maven-plugin` plugin calls the `protobuf` compiler (`protoc`) to generate the stubs classes.
+
+[[build-service]]
+=== Build the Service
+
+After doing the previous steps, you can build the project to generate the stubs. These stubs are needed to resolve the types used in the service implementation.
+
+Example:
+
+[source, shell]
+----
+${webapp root folder} > mvn clean install
+----
+
+The stubs should be generated with similar structure like the following:
+
+image::grpc/grpc-stubs-folders.png[Stubs folders]
+
+target/generated-sources/protobuf/grpc-java/fish.payara.samples.grpc:: Here you can see the service class
+target/generated-sources/protobuf/java/fish.payara.samples.grpc:: Here you can see the types associated to the service
+
+[[extend-service-implementation]]
+=== Extending service implementation class
+
+To implement the service's endpoint you'll have to extend the implementation base class that is nested inside the recently compiled gRPC stub class located under `/target/generated-sources/protobuf/grpc-java`:
+
+[source, Java]
+----
+@Dependent
+public class PayaraService extends PayaraServiceGrpc.PayaraServiceImplBase {
+    private final static Logger log = Logger.getLogger(PayaraService.class.getName());
+    @Override
+    public void communicate(fish.payara.samples.grpc.PayaraReq request,
+                            io.grpc.stub.StreamObserver<fish.payara.samples.grpc.PayaraResp> responseObserver) {
+        final String message = request.getMessage(); //getting message from the request
+        log.info(String.format("Processing message: %s", message)); //printing incoming message from the request
+        responseObserver.onNext(response(message)); //setting the message to the response
+        responseObserver.onCompleted(); //indicating that the response is complete
+    }
+
+    private static final fish.payara.samples.grpc.PayaraResp response(String message) {
+        return fish.payara.samples.grpc.PayaraResp.newBuilder() //creating builder
+                .setMessage(message) //setting response message
+                .build(); //build the response
+    }
+}
+----
+
+Our example is a simple "echo" service that will print a "Hello World" message. We can see that the  `communicate` method receives the `PayaraReq` and the `StreamObserver<fish.payara.samples.grpc.PayaraResp>` parameters which are needed to process the incoming request with the 'Hello World' message and create the response using the same message.
+
+[[deploy-service-application]]
+=== Deploy service application
+
+The final step is to deploy the application in a Payara Server domain. We can do this by using the Admin Console or the Asadmin CLI:
+
```

Comment:
The deploy works only if I remove:
`<!DOCTYPE payara-web-app PUBLIC "-//Payara.fish//DTD Payara Server 4 Servlet 3.0//EN" "https://raw.githubusercontent.com/payara/Payara-Server-Documentation/master/schemas/payara-web-app_4.dtd">`
from web.xml

Reply:
the web.xml is not needed